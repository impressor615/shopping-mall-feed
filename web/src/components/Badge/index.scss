@import "~@/assets/styles/variables";

.badge {
  display: inline-block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  padding: 10px;

  color: $darkgrey;
  font-weight: bold;
  font-size: 25px;

  text-align: center;

  &-gold, &-silver, &-bronze {
    color: $white;
  }

  &-gold {
    background: $gold;
  }

  &-silver {
    background: $silver;
  }

  &-bronze {
    background: $bronze;
  }
}