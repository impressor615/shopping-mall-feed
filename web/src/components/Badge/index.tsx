import "./index.scss";

import classnames from "classnames";
import React from "react";

interface BadgeProps {
  rank: number;
}

const Badge = ({ rank }: BadgeProps) => (
  <span className={classnames(
    "badge",
    { "badge-gold": rank === 1 },
    { "badge-silver": rank === 2 },
    { "badge-bronze": rank === 3 },
  )}>
    { rank }
  </span>
);

export default Badge;
