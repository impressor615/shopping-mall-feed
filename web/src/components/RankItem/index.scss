@import "~@/assets/styles/variables";

.rank-item {
  display: flex;
  text-decoration: none;
  padding: 15px 20px;
  border-bottom: 3px solid $lightgrey;

  img {
    width: 120px;
    height: 120px;
    border-radius: 10px;

    margin-right: 20px;
  }

  .badge {
    margin-top: 10px;
    margin-right: 20px;
  }

  .rank-desc h3 {
    color: $black;
  }

  .rank-desc p {
    color: $secondary;
    font-weight: bold;
  }
}