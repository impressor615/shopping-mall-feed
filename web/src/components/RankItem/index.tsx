import "./index.scss";

import React from "react";

import Badge from "@/components/Badge";

interface RankItemProps {
  rank: number;
  name: string;
  tags: string;
  imgSrc: string;
  url: string;
}

const RankItem = ({ rank, name, tags, imgSrc, url }: RankItemProps) => (
  <a className="rank-item" href={url} target="_blank">
    <Badge rank={rank} />
    <img src={imgSrc} alt="mall-title" />
    <div className="rank-desc">
      <h3>{ name }</h3>
      <p>{ tags }</p>
    </div>
  </a>
);

export default RankItem;
