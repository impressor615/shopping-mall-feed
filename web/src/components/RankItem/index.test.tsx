import enzyme from "enzyme";
import React from "react";

import Badge from "@/components/Badge";
import RankItem from "@/components/RankItem";

describe("RankItem", () => {
  const rank = 1;
  const shopName = "name";
  const shopTags = "tag1, tag2";
  const imgSrc = "img";
  const url = "url";
  it("should render Badge, img, rank-desc", () => {
    const Component = (
      <RankItem
        url={url}
        rank={rank}
        name={shopName}
        tags={shopTags}
        imgSrc={imgSrc}
      />
    );
    const wrapper = enzyme.shallow(Component);
    expect(wrapper.find(Badge)).toHaveLength(1);
    expect(wrapper.find("img")).toHaveLength(1);
    expect(wrapper.find(".rank-desc")).toHaveLength(1);
  });

  it("should be matched with snapshot", () => {
    const Component = (
      <RankItem
        url={url}
        rank={rank}
        name={shopName}
        tags={shopTags}
        imgSrc={imgSrc}
      />
    );
    const wrapper = enzyme.shallow(Component);
    expect(wrapper).toMatchSnapshot();
  });
});
