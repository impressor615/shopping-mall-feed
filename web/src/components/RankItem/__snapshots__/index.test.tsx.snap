// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankItem should be matched with snapshot 1`] = `
<a
  className="rank-item"
  target="_blank"
>
  <Badge
    rank={1}
  />
  <img
    alt="mall-title"
    src="img"
  />
  <div
    className="rank-desc"
  >
    <h3>
      name
    </h3>
    <p>
      tag1, tag2
    </p>
  </div>
</a>
`;
